File: backend/discovery_door/src/main/java/api/discovery_door/Regist.java
package api.discovery_door;

public class Regist {
    private final String date;
    private final int tempoDeViagem;
    private final Museum museu;

    public Regist(String[] info,Museum museu) {
        String[] dateSplit = info[1].split("T");
        
        String[] hourSplit = dateSplit[1].split(":");
        this.date = dateSplit[0] + " " + hourSplit[0] + ":" + hourSplit[1];

        this.tempoDeViagem = Integer.parseInt(info[2]);
        this.museu = museu;
    }

    public String getDate() {
        return this.date;
    }


    public int getTempoDeViagem() {
        return this.tempoDeViagem;
    }


    public Museum getMuseu() {
        return this.museu;
    }

}

File: frontend/discovery_door/lib/data/regist.dart
import 'package:discovery_door/data/museum.dart';

class Regist {
  late String date;
  late int tempoDeViagem;
  late Museum museum;

  Regist({
    required this.date,
    required this.tempoDeViagem,
    required this.museum,
  });

  factory Regist.fromJson(Map<String, dynamic> json) {
    return Regist(
      date: json['date'],
      tempoDeViagem: json['tempoDeViagem'],
      museum: Museum.fromJson(json['museu']),
    );
  }
}
